src/librustc_resolve/lib.rs

```diff
@@ -2470,6 +2493,32 @@ impl<'a, 'crateloader: 'a> Resolver<'a, 'crateloader> {
         self.ribs[TypeNS].pop();
     }
 
+    fn with_tuple_struct_self_ctor_rib<F>(&mut self, self_ty: &Ty, f: F)
+        where F: FnOnce(&mut Resolver)
+    {
+        let variant_def = if self.session.features_untracked().tuple_struct_self_ctor {
```

Comment:
Feature gates very rarely *enable* features, they usually report an error when some feature is already used, so this condition is not necessary.

Reply:
Thanks. Fixed.